### PR TITLE
Improve the chart rendering for the tabular data payloads

### PR DESCRIPTION
**Pull request by emeeks**
- [x] I have read the [Contributor Guide](https://github.com/nteract/nteract/blob/master/CONTRIBUTING.md)

Here's a rough first shot at getting the same functionality seen in the Plotly integration but from [Semiotic](https://github.com/emeeks/semiotic). It's rough, all the code is in one file like the Plotly code was, and the UI elements are garbage, and also there's more functionality in the charts, so any feedback from any perspective (code, dataviz, structure, design) is welcome.

### CODE REVIEW

**Comment by rgbkrk** [comment]
This is the first time we've needed any amount of options/selection UI so it's totally fine this is (self-described as) rough. The main focus is the visualization from rough user-provided data and this is looking pretty darn great from an interactive standpoint!

What I think would be useful, as a project, is if we provided some UI components for options/selection that work well inside the app. Happy to help carve them out in here or as a new PR/new package. I know @theengineear is interested too.

I'm going to be on the road tomorrow so I think what I'll do is make sure I have this cloned / installed so I can try it out.

**Comment by rgbkrk** [comment]
Related issues for this component:

* Most similar issue, bringing us closer to Zeppelin's data viewer: https://github.com/nteract/nteract/issues/2324
* Improving the icons https://github.com/nteract/nteract/issues/2321

**Comment by rgbkrk** [comment]
This is working pretty well inside the showcase!

screen shot 2018-04-03 at 10 32 33 am

On Desktop (and likely the jupyter extension), I'm getting errors:

screen shot 2018-04-03 at 10 33 38 am

**Comment by rgbkrk** [comment]
That might be because we have Immutable.JS structures going in and we're not converting them (?). I'll tinker with it to figure it out.

**Comment by emeeks** [comment]
Without being able to dig into it, it looks like the component isn't getting a `height` prop because it's the SVG itself that isn't drawing with height. There could be other problems but I'd look at that first.

File: packages/transform-dataresource/src/index.js [hunk 1]
```diff
@@ -40,37 +69,389 @@ const DataResourceTransformGrid = ({
...
+    label: "Stacked Area Chart (Percent)"
+  },
+  {
+    type: "bumparea",
+    label: "Ranked Area Chart"
+  }
+];
 
-const plotViewTypes = Object.keys(viewTypeToPlotlyTraceProps);
+const metricDimSelector = (values, selectionFunction, title) => (
```
Comment (rgbkrk) [inline packages/transform-dataresource/src/index.js]: Is this a semiotic pattern where these function signatures are not a props bag but it returns a React.Element?

**Comment by rgbkrk** [comment]
Ok I've snuck in a default height property for semiotic to work with. Still rendering issues for some of the chart types, I'll keep at it. I'd really like to start breaking components up into separate files here.

**Comment by rgbkrk** [comment]
It's amusing how much we ported over from the `infer` code from frictionless tableschema.

**Comment by rgbkrk** [comment]
This is a bit of an aside to this component, likely essential for maintenance though.

We've got some decisions to make about the table schema inference. I recall we forked it internally into here to reduce dependencies originally. It's not currently flow typed (and doesn't match our current style well). The original library we built from is now using an asynchronous interface.

Some thoughts / direction:

* We tear all the inference code out, require that a `schema` is passed in
* We re-write the inference code ourselves
* We move over to the current version of https://github.com/frictionlessdata/tableschema-js, which means we probably need to do it at the top level component, relying on setting `state` internally to the component and potentially memoizing

At the very least, I'd like to group the inference code in a separate module

**Comment by rgbkrk** [comment]
@emeeks is this getting close enough that we can merge the first pass here and then iterate with new PRs? I believe from our in-person work that you also upgraded the semiotic version.

**Comment by rgbkrk** [comment]
Let's keep this moving on forward, thanks @emeeks!

**Pull request merged**